packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -130,16 +139,16 @@ export class PluginDeployerImpl implements PluginDeployer {
                         if (dependencies && !pluginsToDeploy.has(dependencies.metadata.model.id)) {
                             pluginsToDeploy.set(dependencies.metadata.model.id, deployerEntry);
                             if (dependencies.mapping) {
-                                chunk.push(dependencies.mapping);
+                                dependenciesChunk.push(dependencies.mapping);
                             }
                         }
                     }
                 } catch (e) {
                     console.error(`Failed to resolve plugins from '${current}'`, e);
                 }
-            }
-            for (const dependencies of chunk) {
-                for (const [dependency, deployableDependency] of dependencies) {
+            }));
+            for (const dependencies of dependenciesChunk) {
+                for (const [dependency, deployableDependency] of dependencies!) {
```

Comment:
dependencies could be undefined and then you'll get "undefined is not iterable"

Reply:
Only non-undefined values seem to be pushed inside the array, doesn't seem to be any issue here. Typing could be refined earlier maybe.

Reply:
Right. So maybe change line 120 typing to be `Array< Map<string, string>>`

Reply:
and remove ! here

Reply:
Done